packages/core/admin/server/src/services/statistics.ts

```diff
@@ -0,0 +1,425 @@
+  locale: string | null;
+  createdBy: User;
+  updatedBy: User;
+  strapi_assignee?: object | null;
+  strapi_stage?: StrapiStage | null;
+};
+
+export interface PieData {
+  name: string;
+  value: number;
+  entries?: Entry[];
+}
+
+type Locales = {
+  [key: string]: Entry[];
+};
+
+export type KeyNumbers = {
+  [key: string]: number;
+};
+
+interface KeyNumberProvider {
+  name: string;
+  resolve(): Promise<number>;
+}
+
+export type ContentTypeStatistics = {
+  locales: Locales;
+  count: number;
+  modifiedCount: number;
+  statusPieChart: PieData[];
+  localePieChart: PieData[];
+  topContributors: User[];
+  latestDraftEntries: Entry[];
+  latestPublishedEntries: Entry[];
+  stackedAreaChartLocales: Array<{ [key: string]: string }>;
+  stackedAreaChartStages: Array<{ [key: string]: string }>
+  assignedEntriesPieChart: PieData[];
+  assignedEntriesCount: number;
+  stackedAreaChartAssignedStages: Array<{ [key: string]: string }>;
+};
+
+export interface User {
+  id: number;
+  username: string;
+  firstname: string;
+  lastname: string;
+}
+
+export interface Payload {
+  entry: {
+    documentId: string;
+    locale?: string;
+  };
+  uid: string;
+}
+
+export type Activity = {
+  action: string;
+  createdAt: string;
+  date: string;
+  documentId: string;
+  id: number;
+  locale: string;
+  payload: Payload;
+  publishedAt: string;
+  updatedAt: string;
+  user: User;
+};
+
+export type Release = {
+  createdAt: string;
+  documentId: string;
+  id: number;
+  locale: string;
+  name: string;
+  publishedAt: string;
+  releasedAt: string;
+  scheduledAt: string;
+  status: string;
+  timezone: string;
+  updatedAt: string;
+};
+
+export type EEStatistics = {
+  upcomingReleases: Release[] | Array<object>;
+  latestActivities: Activity[] | Array<object>;
+  upcomingReleasesCount: number,
+  workflowsCount: number,
+};
+
+const providers: Map<string, KeyNumberProvider> = new Map();
+
+export function registerKeyNumbersProvider(provider: KeyNumberProvider): void {
+  if (providers.has(provider.name)) {
+    throw new Error(`Provider with name ${provider.name} already exists`);
+  }
+
+  providers.set(provider.name, provider);
+}
+
+export async function getKeyNumbers(): Promise<KeyNumbers> {
+  const keyNumbers: KeyNumbers = {
+    entries: 0,
+    assets: 0,
+    contentTypes: 0,
+    components: 0,
+    locales: 0,
+    admin_users: 0,
+    webhooks: 0,
+    apiTokens: 0
+  };
+
+  const results = Array.from(providers.values()).map((provider) =>
+    provider.resolve().then((result) => {
+      keyNumbers[provider.name] = result;
+    })
+  );
+
+  await Promise.all(results);
+
+  return keyNumbers;
+}
+
+export async function getContentTypeStatistics(contentType: UID.CollectionType, userId: number): Promise<ContentTypeStatistics> {
+  // Fetch entries with associated user data that would always be there and just enough for the dashboard
+  const [entries, initialCount] = await strapi.db.query(contentType).findWithCount({
```

Comment:
rbac rules should be checked here to ensure we are not leaking any protected field information (Names could be protected with RBAC rules and this would not respect that) + permissions should be check to see if I am allowed to know that the content-type exists

In its current state an user with 0 permissions and just a jwt token can see firstnames of everyone plus can check any content type like users to see how many users exist and when they where created.

Reply:
+1 , normally we do add those set of permissions in the controller handler, we should check if user has permissions to read the content type by its uid.

an example of how you would build such permission check on a controller:
```ts
	const { userAbility } = ctx.state;
    if (userAbility.cannot('plugin::content-manager.explorer.read', uid) {
      return ctx.forbidden()
    }
```